2023-09-18 17:41:34.221052: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:41:34.221216: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:41:41,817 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:41:41,900 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:41:41,900 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:41:41,900 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:41:41,937 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:41:41,938 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:41,956 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.46}
INFO flwr 2023-09-18 17:41:41,956 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:41:42,076 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,114 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:41:42,116 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,124 | server.py:125 | fit progress: (1, 0.3897330089703304, {'accuracy': 0.84}, 0.16746909999999993)
DEBUG flwr 2023-09-18 17:41:42,124 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,157 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:41:42,157 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:41:42,157 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,182 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,192 | server.py:125 | fit progress: (2, 0.39087403297473905, {'accuracy': 0.84}, 0.23519950000000023)
DEBUG flwr 2023-09-18 17:41:42,192 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,209 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,209 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,236 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,247 | server.py:125 | fit progress: (3, 0.3908635150543058, {'accuracy': 0.84}, 0.2911988000000001)
DEBUG flwr 2023-09-18 17:41:42,248 | server.py:173 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:41:42,263 | server.py:187 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,264 | server.py:223 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:41:42,288 | server.py:237 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,300 | server.py:125 | fit progress: (4, 0.39184887256495654, {'accuracy': 0.84}, 0.3435182000000001)
DEBUG flwr 2023-09-18 17:41:42,300 | server.py:173 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:41:42,323 | server.py:187 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,323 | server.py:223 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2023-09-18 17:41:42,359 | server.py:237 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,372 | server.py:125 | fit progress: (5, 0.39178024085659824, {'accuracy': 0.84}, 0.41512490000000035)
DEBUG flwr 2023-09-18 17:41:42,372 | server.py:173 | evaluate_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,396 | server.py:187 | evaluate_round 5 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,396 | server.py:223 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,432 | server.py:237 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,444 | server.py:125 | fit progress: (6, 0.3915239159213894, {'accuracy': 0.84}, 0.48721749999999986)
DEBUG flwr 2023-09-18 17:41:42,444 | server.py:173 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,468 | server.py:187 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,469 | server.py:223 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,502 | server.py:237 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,514 | server.py:125 | fit progress: (7, 0.3915218756373073, {'accuracy': 0.84}, 0.5577763999999998)
DEBUG flwr 2023-09-18 17:41:42,514 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,546 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,546 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,581 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,593 | server.py:125 | fit progress: (8, 0.3915218314718139, {'accuracy': 0.84}, 0.6361551999999993)
DEBUG flwr 2023-09-18 17:41:42,593 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,628 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,628 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,664 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,677 | server.py:125 | fit progress: (9, 0.39152183000835217, {'accuracy': 0.84}, 0.7201371999999999)
DEBUG flwr 2023-09-18 17:41:42,677 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,707 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:41:42,707 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,742 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:41:42,756 | server.py:125 | fit progress: (10, 0.39152182995492507, {'accuracy': 0.84}, 0.7996958000000003)
DEBUG flwr 2023-09-18 17:41:42,756 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:41:42,784 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:41:42,784 | server.py:147 | FL finished in 0.8280139000000002
INFO flwr 2023-09-18 17:41:42,785 | app.py:218 | app_fit: losses_distributed [(1, 0.39497816811005276), (2, 0.393903162330389), (3, 0.3939003348350525), (4, 0.3953663229942322), (5, 0.39543965927192143), (6, 0.3948304138013295), (7, 0.3948302860770907), (8, 0.3948302728789193), (9, 0.3948302728789193), (10, 0.3948302728789193)]
INFO flwr 2023-09-18 17:41:42,785 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:41:42,785 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:41:42,785 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.3897330089703304), (2, 0.39087403297473905), (3, 0.3908635150543058), (4, 0.39184887256495654), (5, 0.39178024085659824), (6, 0.3915239159213894), (7, 0.3915218756373073), (8, 0.3915218314718139), (9, 0.39152183000835217), (10, 0.39152182995492507)]
INFO flwr 2023-09-18 17:41:42,785 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.46), (1, 0.84), (2, 0.84), (3, 0.84), (4, 0.84), (5, 0.84), (6, 0.84), (7, 0.84), (8, 0.84), (9, 0.84), (10, 0.84)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99406716,  0.96745868,  0.74212551,  0.51935457,  0.25106775,
         0.08306541, -0.06436537, -0.01711919]]), array([0.04970143])]
[array([[-1.99406716,  0.96745868,  0.74212551,  0.51935457,  0.25106775,
         0.08306541, -0.06436537, -0.01711919]]), array([0.04970143])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01845999,  1.01537741,  0.77739122,  0.52078626,  0.26793922,
         0.10354993, -0.0485759 ,  0.09388812]]), array([-0.17319699])]
[array([[-2.01845999,  1.01537741,  0.77739122,  0.52078626,  0.26793922,
         0.10354993, -0.0485759 ,  0.09388812]]), array([-0.17319699])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01922551,  1.01627755,  0.77782327,  0.52114954,  0.26803884,
         0.10398937, -0.04848671,  0.09421599]]), array([-0.17544381])]
[array([[-2.01922551,  1.01627755,  0.77782327,  0.52114954,  0.26803884,
         0.10398937, -0.04848671,  0.09421599]]), array([-0.17544381])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01546676,  1.00965294,  0.7958553 ,  0.46737784,  0.26936407,
         0.13187356, -0.05895831,  0.04151173]]), array([-0.08737103])]
[array([[-2.01546676,  1.00965294,  0.7958553 ,  0.46737784,  0.26936407,
         0.13187356, -0.05895831,  0.04151173]]), array([-0.08737103])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.01880679,  1.0114683 ,  0.79636121,  0.46781021,  0.26941389,
         0.13158159, -0.05965359,  0.04243524]]), array([-0.08725355])]
[array([[-2.01880679,  1.0114683 ,  0.79636121,  0.46781021,  0.26941389,
         0.13158159, -0.05965359,  0.04243524]]), array([-0.08725355])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05397819,  1.0640471 ,  0.82676128,  0.43365714,  0.21294505,
         0.10220323, -0.05980183,  0.02207462]]), array([0.00740028])]
[array([[-2.05397819,  1.0640471 ,  0.82676128,  0.43365714,  0.21294505,
         0.10220323, -0.05980183,  0.02207462]]), array([0.00740028])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05407916,  1.06391796,  0.82659608,  0.43368996,  0.21292909,
         0.10224254, -0.05980514,  0.02204463]]), array([0.00785406])]
[array([[-2.05407916,  1.06391796,  0.82659608,  0.43368996,  0.21292909,
         0.10224254, -0.05980514,  0.02204463]]), array([0.00785406])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05408192,  1.06391976,  0.82659793,  0.43369281,  0.21292889,
         0.10224193, -0.05980399,  0.02204486]]), array([0.0078512])]
[array([[-2.05408192,  1.06391976,  0.82659793,  0.43369281,  0.21292889,
         0.10224193, -0.05980399,  0.02204486]]), array([0.0078512])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05408214,  1.06391983,  0.82659798,  0.43369282,  0.212929  ,
         0.10224196, -0.05980401,  0.02204484]]), array([0.00785124])]
[array([[-2.05408214,  1.06391983,  0.82659798,  0.43369282,  0.212929  ,
         0.10224196, -0.05980401,  0.02204484]]), array([0.00785124])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.05408215,  1.06391983,  0.82659798,  0.43369282,  0.212929  ,
         0.10224196, -0.05980402,  0.02204484]]), array([0.00785124])]
[array([[-2.05408215,  1.06391983,  0.82659798,  0.43369282,  0.212929  ,
         0.10224196, -0.05980402,  0.02204484]]), array([0.00785124])]
>>>>>>>>>>>>>run time:
0.640625s
